Autodetect Feature, effectively fixes #4 and #10

Mystro256 opened the pull request:
After seeing #10 (and originally #4), I wrote a little fix to make the extension automatically check if the battery menu is there, if not it uses the accessibility menu. Also taking into account that the user may have this disabled, I also made it use the usermenu instead if a11y is not available.

A much needed fix for sure :)

Mystro256 committed:
Add autodetect feature, effectively fixing #4 and #10
Checks if theres a battery menu, if not tries a11y, if all else fails it
resorts to using userMenu

RaphaelKimmig commented:
line 33 looks like a typo to me?
I'll test the patch as soon as I can and then merge it. 
Thank you very much for sharing your improvements!

Mystro256 commented:
Ih yeah, sorry i was in a rush when I committed it. The whole line 33 is unnecessary, as it was just a line I used in testing the functionality of the changes.

Good eye, I'll mark it.

Mystro256 commented on presentationmode@ampad.de/extension.js:
```diff
@@ -22,14 +22,30 @@ let SessionProxy = DBus.makeProxyClass(SessionIface);
+    let temp = Main.panel._statusArea.battery._deviceItems;
+    if(temp == "")
+    {//check for no battery or power device, i.e. no power menu
+        lettemp = new Main.Panel.STANDARD_STATUS_AREA_SHELL_IMPLEMENTATION["a11y"]
```
Line 33 should be removed :)

Mystro256 committed:
Fixed Typo and added modularity

Fixed the lettemp typo and added modularity by determining the menu type on enable() rather than on initialization (i.e. in case noally is enabled prior to enabling presentation mode). I also added some comments.

Mystro256 commented:
I added some more improvements! :)
I just forgot to remove that line, I'll do that now.

Mystro256 committed:
Removed redundant line of code

I forgot to removed this in my last commit ;)

Mystro256 commented:
Also, I did some testing; this seems to work just fine on several laptops and desktops (with and without the a11y menu). The only issue I've seen is that this does not work with "gnome-shell-reflection" (https://github.com/gecos-team/gnome-shell-reflection) unless noa11y is also enabled at the same time (the no-a11y functionality built in is a little broken)

I'm half tempted to fork it, fix it up, and remove that no-a11y functionality, so I can request a fedora package for it. ;)

Mystro256 committed:
Rebranding to inhibit-applet and some fixes

Fixed locale issues, also added french translation, new translations welcome.

Mystro256 committed:
Fixing up Readme file

Mystro256 committed:
Added a screenshot

Mystro256 committed:
Added a nicer readme

Mystro256 committed:
Fixed screenshot in readme

Mystro256 committed:
Removed a redundant line

Mystro256 committed:
Added power/french screenshot

Mystro256 committed:
Added some aesthetic appeal to the screenshots

Mystro256 committed:
Typo in the last commit

Mystro256 committed:
Added polish translation thanks to belegdol :)

lavia committed:
Hebrew locale

Mystro256 committed:
Merge pull request #3 from lavi741/master

Added Hebrew locale thanks to lavi741 :)

Mystro256 committed:
Should Fix Issue #1

Checks for the batterymenu visibility more directly

Mystro256 committed:
Code clean up and a hack to fix issue #1

Mystro256 committed:
Fix the power/french screenshot

Mystro256 committed:
Now uses it's own menu with changing icon

This seems to just make more sense, plus it avoids using a hack to check the battery menu visibility and allows for a dynamic icon to check status easily

Mystro256 committed:
Updated Readme

Forgot to change the description

Mystro256 commented:
This seems broken and have no interest in fixing it
